sqlitecluster/SQLiteNode.cpp

```diff
@@ -2274,3 +2194,327 @@ bool SQLiteNode::peekPeerCommand(SQLiteNode* node, SQLite& db, SQLiteCommand& co
+
+        // The "success" case here duplicates part of "SQLite::prepare" because we need this data before it's ready in the DB.
+        response["NewHash"] = success ? SToHex(SHashSHA1(db.getCommittedHash() + db.getUncommittedQuery())) : commitHash;
+        response["ID"] = message["ID"];
+        Peer* masterPeer = node._masterPeer;
+        if (masterPeer) {
+            node._sendToPeer(masterPeer, response);
+        } else {
+            SWARN("Lost master while committing, can't send response.");
+            db.rollback();
+            return false;
+        }
+        SINFO("Sent " << verb << " for QUORUM transaction.");
+    }
+
+    // Just do some logging.
+    uint64_t transitTimeUS = slaveDequeueTimestamp - masterSentTimestamp;
+    uint64_t applyTimeUS = STimeNow() - slaveDequeueTimestamp;
+    float transitTimeMS = (float)transitTimeUS / 1000.0;
+    float applyTimeMS = (float)applyTimeUS / 1000.0;
+    SINFO("Replicated transaction " << message.calcU64("NewCount") << ", sent by master at " << masterSentTimestamp
+          << ", transit/dequeue time: " << transitTimeMS << "ms, applied in: " << applyTimeMS << "ms, should COMMIT next.");
+
+    // Now we wait. For quorum commands, we wait for a server response (or disconnection).
+    // For non-quorum commands, we wait for all previous commits to succeed, so that we know we commit in order.
+    while (!node._workersShouldFinish.load()) {
+        string quorumString = useQuorum ? "QUORUM" : "non-QUORUM";
+        SINFO("Waiting for " << db.getCommitCount() << " == " << (commitNum - 1) << " and " << node._safeCommitTarget.load() << " >= " << commitNum << " in " << quorumString << " commit.");
+
+        // Grab our lock so nothing changes while we check our conditions.
+        unique_lock<mutex> lock(node._workReadyMutex);
+
+        // See if we've lost our master entirely.
+        if (node._safeCommitTarget.load() < startingCommitTarget) {
+            SWARN("Master lost mid-commit. Discarding transaction and starting over.");
+            db.rollback();
+            return true;
+        }
+
+        // If this is a QUORUM transaction, and there's no master, or master changed since we started, we abort.
+        // For non-quorum transactions, we ignore this condition, as it's possible to synchronize with no node
+        // currently mastering.
+        if (useQuorum && (!startingMaster || (startingMaster != node._masterPeer))) {
+            SWARN("No master or master changed mid-QUORUM transaction. Aborting.");
+            db.rollback();
+            return true;
+        }
+
+        // What if the server rolled back the transaction because not enough peers approved it?
+        if (commitNum == node._rollbackTransactionID.load()) {
+            // Reset the value in case something else gets rolled back in the future, and back out.
+            SALERT("Rolling back failed QUORUM transaction.");
+            node._rollbackTransactionID.store(UINT64_MAX);
```

Comment:
i read the comment in the header and still don't totally understand, what is this and why are we storing this static value in it? any significance to this value?